homeassistant/components/prowl/__init__.py

```diff
@@ -1 +1,54 @@
 """The prowl component."""
+
+import logging
+
+import prowlpy
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN, PLATFORMS
+from .notify import PLATFORM_SCHEMA, ProwlNotificationEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+__all__ = ["DOMAIN", "PLATFORM_SCHEMA"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a Prowl service."""
+    # This just checks if the API key is valid before initialising the entry
+    prowl = ProwlNotificationEntity(hass, entry.title, entry.data[CONF_API_KEY])
+    try:
+        if not await prowl.async_verify_key():
+            raise ConfigEntryAuthFailed(
+                "Unable to validate Prowl API key (Key invalid or expired)"
+            )
+    except TimeoutError as ex:
+        raise ConfigEntryNotReady("API call to Prowl failed") from ex
+    except prowlpy.APIError as ex:
+        if str(ex).startswith("Not accepted: exceeded rate limit"):
+            raise ConfigEntryNotReady(
+                "Prowl API rate limit exceeded, try again later"
+            ) from ex
+        if str(ex).startswith(
+            "Not approved: The user has yet to approve your retrieve request"
+        ):
+            raise ConfigEntryNotReady(
+                "Prowl user has yet to approve your retrieve request"
+            ) from ex
```

Comment:
When does this happen? Do we check it in the config flow? If we do, can the approval be revoked so we need to check it here too?
Should we inform the user via a repair in a follow-up?

Reply:
My mistake of implementing this. The API key approval is if you use a Provider Key feature in the API (which isn't something that we've implemented here). I've removed the references to this.